hail/python/test/hail/table/test_table.py

```diff
@@ -1216,6 +1216,15 @@ def test_repartition_empty_key(self):
         assert ht.naive_coalesce(4)._same(ht)
         assert ht.repartition(3, shuffle=False)._same(ht)
 
+    def test_path_collision_error(self):
+        path = new_temp_file(suffix='ht')
+        ht = hl.utils.range_table(10)
+        ht.write(path)
+        ht = hl.read_table(path)
+        with pytest.raises(hl.utils.FatalError) as exc:
+            ht.write(path)
+        assert "both an input and output source" in str(exc.value)
```

Comment:
I think you can do something like `self.assertRaises(error, "regex match")` to bundle that statement together---are we moving away from that sort of assertion?

Reply:
🤷‍♂ we're using pytest now, so pytest is probably preferable to unittest. either one will work though